File: 1. Week 1 - Barchart/exercise/R/visualization.R
# Visualize Distance per Month using ggplot2

# Load Libraries
if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# 1. Load Data
# We need to make sure the path points to our data file
data_path <- "../../data/distance_per_month.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # 2. Plot
  # TODO: Create the bar chart using ggplot2
  # Use geom_col() for the bars
  # Add labels and a theme
  
  # p <- ggplot(df, aes(x = Month, y = TotalDistance)) + ...
    
  
  # 3. Save
  # Uncomment the following lines once 'p' is created
  # ggsave("monthly_distance_chart.png", plot = p, width = 10, height = 6)
  # print("Chart saved to monthly_distance_chart.png")
  
} else {
  print(paste("Error: File not found at", data_path))
}

File: 1. Week 1 - Barchart/exercise/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    # Load data
    data_path = '../../data/distance_per_month.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load the dataset using pandas
    # df = ...
    
    # 1. Plot
    plt.figure(figsize=(10, 6))
    # TODO: Create a bar chart using plt.bar()
    # plt.bar(...)
    
    # 2. Labels
    # TODO: Add title, x-label, and y-label
    # plt.title(...)
    # plt.xlabel(...)
    # plt.ylabel(...)
    
    plt.xticks(rotation=45)
    plt.tight_layout()
    
    # 3. Save
    # TODO: Save the figure to a file
    # plt.savefig(...)
    
    print("Chart saved to monthly_distance_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()
